internal-client/src/main/java/org/apache/uniffle/client/factory/ShuffleServerClientFactory.java

```diff
@@ -75,9 +80,30 @@ public synchronized ShuffleServerClient getShuffleServerClient(
     return serverToClients.get(shuffleServerInfo);
   }
 
+  public synchronized void cleanUselessShuffleServerClients(
```

Comment:
Is is possible that the another thread to hold the removing client?

Reply:
I think it is impossible, if another thread to hold the removing client, the `shuffleServerInfo` of the removing client will always be in `shuffleServerInfoMap`.